basex-core/src/main/java/org/basex/core/locks/DBLocking.java

```diff
@@ -138,7 +138,9 @@ public void acquire(final Proc pr, final StringList read, final StringList write
       }
       // global read locking
       if(read == null) {
-        while(localWriters > 0) {
+        while(localWriters > 0 &&
+            // We're the only writer, allow global read lock anyway
+            !(1 == localWriters && !(null == write || write.isEmpty()))) {
```

Comment:
Well, it was quite a challenge for me to correctly decode the negated conditions ;) But as long as it's correct, it should be fine.

By the by, I am indeed afraid that I probably got it wrong. This is my updated attempt… The loop will be interrupted (e.g. `globalLock.wait()` will not be called anymore) if:

1. `localWriters` = `0`, or
2. `localWriters` = `1` and if there is at least one local write lock

Do you agree? I'm asking to ensure that the code is really correct, and that all cases are correctly covered.